c/zos.c

```diff
@@ -275,7 +278,7 @@ OTCB *getOTCB(void) {
 
 ASCB *getASCB(void) {
   int *mem = (int*)0;
-  return (ASCB*)(mem[CURRENT_ASCB/sizeof(int)]&0x7FFFFFFF);
+  return (ASCB*)INT2PTR(mem[CURRENT_ASCB/4]);
```

Comment:
Can the ASCB address ever have the highest bit set to 1? Asking because the old code used to and with `0x7fffffff`. Perhaps, that was there for a reason.

Reply:
Should not matter for ASCB according to experts.